shell/platform/windows/system_utils.cc

```diff
@@ -23,22 +24,49 @@ std::vector<LanguageInfo> GetPreferredLanguageInfo() {
   return language_info;
 }
 
-std::vector<std::wstring> GetPreferredLanguages() {
+std::vector<std::wstring> GetPreferredLanguages(
+    const WindowsRegistry& registry) {
   std::vector<std::wstring> languages;
+  BOOL languages_from_registry = TRUE;
+  ULONG buffer_size = 0;
+  ULONG count = 0;
   DWORD flags = MUI_LANGUAGE_NAME | MUI_UI_FALLBACK;
 
-  // Get buffer length.
-  ULONG count = 0;
-  ULONG buffer_size = 0;
-  if (!::GetThreadPreferredUILanguages(flags, &count, nullptr, &buffer_size)) {
-    return languages;
+  // Determine where languages are defined and get buffer length
+  if (registry.GetRegistryValue(HKEY_CURRENT_USER, kGetPreferredLanguageRegKey,
+                                kGetPreferredLanguageRegValue,
+                                RRF_RT_REG_MULTI_SZ, NULL, NULL,
+                                &buffer_size) != ERROR_SUCCESS) {
+    languages_from_registry = FALSE;
+    if (!::GetThreadPreferredUILanguages(flags, &count, nullptr,
+                                         &buffer_size)) {
+      return languages;
+    }
   }
 
-  // Get the list of null-separated languages.
+  // Multi-string must be at least 3-long if non-empty,
+  // as a multi-string is terminated with 2 nulls.
+  //
+  // See:
+  // https://learn.microsoft.com/windows/win32/sysinfo/registry-value-types
+  if (buffer_size < 3) {
+    languages_from_registry = FALSE;
```

Comment:
Will this work as expected? What if `GetThreadPreferredUILanguages` needs a bigger buffer than `3`?